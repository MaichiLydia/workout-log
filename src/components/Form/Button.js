import React from 'react';
import PropTypes from 'prop-types';


function Button(props){
    let classes = 'form__btn'
    if(props.disabled){
        classes += ' form__btn--disabled'
        return (
            <button className={classes} disabled>
                {props.children}
            </button>
        )
    }
    return (
        <button className={classes}>
            {props.children}
        </button>
    )
    
}
Button.propTypes = {
    disabled: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.bool
    ]).isRequired,
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node
    ]).isRequired 
}


export default Button;