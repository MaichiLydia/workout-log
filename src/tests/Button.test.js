import React from 'react';
import Button from '../components/Form/Button';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

it('check render a disabled button', () => {
    const props = {
        disabled: true,
        children: "Click me !"
    }
    const button = mount(<Button {...props}></Button>)

    expect(toJson(button)).toMatchSnapshot()

});

it('should render an active button', () => {
    const props = {
        disabled: false,
        children: "Click me !"
    }
    const button = mount(<Button {...props}></Button>)

    expect(toJson(button)).toMatchSnapshot()

});