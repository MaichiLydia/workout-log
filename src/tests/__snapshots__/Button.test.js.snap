// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check render a disabled button 1`] = `
<Button
  disabled={true}
>
  <button
    className="form__btn form__btn--disabled"
    disabled={true}
  >
    Click me !
  </button>
</Button>
`;

exports[`should render an active button 1`] = `
<Button
  disabled={false}
>
  <button
    className="form__btn"
  >
    Click me !
  </button>
</Button>
`;
